pennylane/templates/swap_networks/ccl2.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the TwoLocalSwapNetwork template.
+"""
+
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+from pennylane.ops import SWAP, FermionicSWAP
+
+
+class TwoLocalSwapNetwork(Operation):
+    r"""Apply two-local gate operations using a canonical 2-complete linear (2-CCL) swap network.
+
+    Args:
+        wires (Iterable or Wires): ordered sequence of wires on which the swap network acts
+        acquaintances (Callable): callable `func(index, wires, param=None, **kwargs)` that returns a two-local operation applied on a pair of logical wires specified by `index` currently stored in physical wires provided by `wires` before they are swapped apart. Parameters for the operation are specified using `param`, and any additional keyword arguments for the callable should be provided using the ``kwargs`` separately
```

Comment:
Few points about this arg:
1. Is the best name for this argument?
2. Both wires and index are needed?
3. Does it need to be a callable? Can it be something simpler?

Reply:
1. The name comes from the paper [arXiv:1905.05118](https://arxiv.org/abs/1905.05118), so I am inclined towards using it.
2. Having access to both allows users to have more freedom in building their acquaintances. 
3. I made them callable to allow users to have as complex acquaintances as they would like to have. Otherwise, to make it simpler, I will have to try to fix the structure myself or increase the complexity on the circuit implementation side, which I want to avoid.